server/pairing/client.go

```diff
@@ -36,17 +36,37 @@ type BaseClient struct {
 
 // NewBaseClient returns a fully qualified BaseClient from the given ConnectionParams
 func NewBaseClient(c *ConnectionParams) (*BaseClient, error) {
-	u, err := c.URL()
-	if err != nil {
-		return nil, err
+
+	var url *url.URL
+	var serverCert *x509.Certificate
+	var certErrs error
+
+	for i := range c.netIPs {
+		u, err := c.URL(i)
+		if err != nil {
+			return nil, err
+		}
+
+		serverCert, err = getServerCert(u)
+		if err != nil {
+			certErrs = fmt.Errorf("%wconnecting to '%s' failed: %w; ", certErrs, u, err)
```

Comment:
Nice use of wrapped errors.

Reply:
At first I used `errors.Join`, but Go version is older than `1.21` in CI 🙁

Reply:
Yeah, I much prefer `errors.Join` but it is too new